pyanaconda/ui/tui/tuiobject.py

```diff
@@ -16,13 +16,69 @@
 # License and may only be used or replicated with the express permission of
 # Red Hat, Inc.
 #
+
 from pyanaconda.ui import common
 
 from pyanaconda import iutil
+from pyanaconda.constants import PASSWORD_CONFIRM_ERROR_TUI, PASSWORD_WEAK_WITH_ERROR, PASSWORD_WEAK, PW_ASCII_CHARS
 from pyanaconda.constants_text import IPMI_ABORTED
+from pyanaconda.i18n import _
+from pyanaconda.users import validatePassword, cryptPassword
+
+from simpleline.render.adv_widgets import ErrorDialog, GetInputScreen, GetPasswordInputScreen, YesNoDialog
+from simpleline.render.screen import UIScreen, Prompt
+from simpleline.render.screen_handler import ScreenHandler
+
+
+def reporting_callback(message):
+    """Callback used for general reporting from acceptance conditions.
+
+    See: `AskUserInput` and `AskPassword` classes.
+    """
+    print(_(message))
+
+
+def report_if_failed(message):
+    """Decorator function to call reporting function on failed condition.
+
+    :param message: Error message which will be printed if condition function fails.
+    :type message: str
+    """
+    def outer_wrapper(f):
+        def wrapper(*args):
+            if not f(*args):
+                report_func = args[-1]
+                report_func(message)
+                return False
+            return True
+
+        return wrapper
+
+    return outer_wrapper
+
+
+def report_check_func():
```

Comment:
I think that eventually we will rewrite the functions that return tuples, because they should accept the reporting function, but this is ok for now.

Reply:
Good idea and yes that is more for future work.